src/main/java/com/tira/restaurants/authlisteners/AuthenticationFailureListener.java

```diff
@@ -0,0 +1,31 @@
+package com.tira.restaurants.authlisteners;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.authentication.event.AuthenticationFailureBadCredentialsEvent;
+import org.springframework.security.web.authentication.WebAuthenticationDetails;
+import org.springframework.stereotype.Component;
+
+import com.tira.restaurants.service.LoginAttemptService;
+
+@Component
+public class AuthenticationFailureListener implements ApplicationListener<AuthenticationFailureBadCredentialsEvent> {
+ 
```

Comment:
I added this AuthenticationFailureListener to block user by IP if he made over 5 attempts with invalid data. But since I will not be blocking users by IP I will also probably delete this part.

Reply:
Yep, this needs to be based on username